File: app/src/main/java/com/gazilla/mihail/gazillaj/presentation/reserve/ReserveView.java
package com.gazilla.mihail.gazillaj.presentation.reserve;

import com.arellomobile.mvp.MvpView;
import com.gazilla.mihail.gazillaj.utils.POJO.Reserve;

public interface ReserveView extends MvpView {

    void inputUserInfo(String name, String phone);
    void putReserve(Reserve reserve, Boolean preorder);
    void resultReserve(String result);
    void showWorningDialog(String err);
    void showLoadingDialog();
    void clouseAppDialog();
}

File: app/src/main/java/com/gazilla/mihail/gazillaj/kotlin/views/ReserveView.kt
package com.gazilla.mihail.gazillaj.kotlin.views

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(value = AddToEndSingleStrategy::class)
interface ReserveView : MvpView {

    fun setUserInfo(name: String, phone: String)
    fun showClWithMyReserve(show: Boolean)

    fun setTimeReserve(time: String)
    fun setDateReserve(date: String)

    fun deleteMyReserve()

    fun showMessageDialog(message: String)
    fun showDeleteMyReserveDialog(message: String)

    fun showReserveSuccessDialog(message: String)
    fun closeReserveActivity()
}